aws_lambda_powertools/utilities/parameters/appconfig.py

```diff
@@ -66,7 +68,7 @@ def __init__(
 
         config = config or Config()
         self.client = boto3.client("appconfig", config=config)
-        self.application = application or os.getenv("POWERTOOLS_SERVICE_NAME") or "application_undefined"
+        self.application = application or os.getenv(constants.SERVICE_NAME_ENV, "service_undefined")
```

Comment:
You can use the new function we created to enforce explicit over env var, etc. 

```suggestion
        self.application = resolve_env_var_choice(choice=application, env=os.getenv(constants.SERVICE_NAME_ENV, "service_undefined"))
```